docs/source/distributed/worker-schedule.rst

```diff
@@ -1,10 +1,16 @@
 Execution in Worker
 ===================
-When an operand is being executed in a worker, it will first allocate memory.
-Then data from other workers or from files already spilled to disk are loaded.
-After that all data required are in memory and calculation can start. When
-calculation is done, the worker then put the result into shared memory cache.
-These four states can be seen in the graph below.
+A Mars worker consists of multiple processes to reduce the impact of the
+notorious global interpreter lock (GIL) in Python. Executions run in separate
+processes. To reduce unnecessary memory copy and inter-process communication,
+shared memory is used to store computation results.
+
+When an operand is submitted into a worker, it will first be queued and wait
+for memory allocation.  When the memory is allocated, data from other workers
```

Comment:
The dependencies are not mentioned here, `dependencies' data either from other workers or from files already spilled to disk are fetched and loaded` seems clarify this.